data/src/main/scala/ch.epfl.scala.index.data/bintray/UpdateBintraySbtPlugins.scala

```diff
@@ -0,0 +1,165 @@
+  private val dateTimeFormatter = DateTimeFormatter.ISO_DATE_TIME
+  private val moduleDescriptorParser = XmlModuleDescriptorParser.getInstance()
+
+  private val baseSubject = "sbt"
+  private val baseRepo = "sbt-plugin-releases"
+
+  /** Matches an ivy.xml file of an sbt-plugin release
+   * <organization>/<artifact>/scala_<version>/sbt_<version>/<version>/ivys/ivy.xml
+   */
+  private val sbtPluginPathRegex: Regex =
+    "(.+?)/(.+?)/scala_(.+?)/sbt_(.+?)/(.+?)/ivys/ivy.xml".r
+
+  def update(): Future[Unit] = {
+    val lastUpdate = parse(Files.readAllLines(lastDownloadPath).get(0))
+
+    val oldReleasesF = Future(sbtPluginRepo.read())
+    val newReleasesF = getNewReleases(lastUpdate)
+
+    for {
+      oldReleases <- oldReleasesF
+      newReleases <- newReleasesF
+    } yield {
+      sbtPluginRepo.update(oldReleases, newReleases)
+      val now = format(OffsetDateTime.now(ZoneOffset.UTC))
+      Files.write(lastDownloadPath, Seq(now).asJava)
+    }
+  }
+
+  private def getNewReleases(
+      lastUpdate: OffsetDateTime
+  ): Future[Seq[SbtPluginReleaseModel]] = {
+    for {
+      packageNames <- bintray.getAllPackages(baseSubject, baseRepo)
+
+      _ = println(s"found ${packageNames.size} packages")
+      allPackages <- Future.traverse(packageNames)(
+        bintray.getPackage(baseSubject, baseRepo, _)
+      )
+
+      packagesToUpdate = allPackages.filter(
+        p => parse(p.updated).isAfter(lastUpdate)
+      )
+      packagesByRepo = packagesToUpdate.groupBy(p => (p.owner, p.repo))
+
+      sbtPlugins <- Future
+        .traverse(packagesByRepo) {
+          case ((owner, repo), packages) =>
+            getSbtPlugins(owner, repo, packages, lastUpdate)
+        }
+        .map(_.flatten.toSeq)
+
+    } yield sbtPlugins
+  }
+
+  private def getSbtPlugins(
+      owner: String,
+      repo: String,
+      packages: Seq[BintrayPackage],
+      lastUpdate: OffsetDateTime
+  ): Future[Seq[SbtPluginReleaseModel]] = {
+    println(s"loading ${packages.size} from $owner/$repo")
+    val packageNames = packages.map(_.name).toSet
+    val createdAfter = format(lastUpdate)
+    for {
+      allIvyFiles <- bintray.searchFiles(owner, repo, "ivy.xml", createdAfter)
+      filteredIvyFiles = allIvyFiles
+        .filter(file => packageNames.contains(file.`package`))
+        .filter(file => parse(file.created).isAfter(lastUpdate))
+      sbtPlugins <- Future
+        .traverse(filteredIvyFiles)(
+          file => downloadPluginDescriptor(owner, repo, file)
+        )
+        .map(_.flatten)
+    } yield sbtPlugins
+  }
+
+  private def downloadPluginDescriptor(
+      owner: String,
+      repo: String,
+      file: BintraySearch
+  ): Future[Option[SbtPluginReleaseModel]] = {
+    file.path match {
+      case sbtPluginPathRegex(org,
+                              artifact,
+                              scalaVersion,
+                              sbtVersion,
+                              version) =>
+        Future {
```

Comment:
Out of curiosity: why do we want to do that in a separate execution thread?

Reply:
Mostly because the `bintray.downloadUrl(...)` at line 117 is doing IO that we want to parallelize (for each new release).